implementations/rust/ockam/ockam_command/src/service/start.rs

```diff
@@ -64,13 +64,11 @@ pub enum StartSubCommand {
         /// The local address of the service
         #[arg(long, default_value_t = kafka_consumer_default_addr())]
         addr: String,
-        /// The address where to bind and where the client will connect to
-        #[arg(long, default_value_t = [127, 0, 0, 1].into())]
-        bootstrap_server_ip: Ipv4Addr,
-        /// The port to bind to and where the client will connect to
+        /// The address where to bind and where the client will connect to alongside its port, <address>:<port>.
+        /// In case just a port is specified, the default loopback address will be used
         #[arg(long)]
-        bootstrap_server_port: u16,
-        /// Local port range dynamically allocated to kafka brokers, mut not overlap with the
+        bootstrap_server: String,
```

Comment:
We should still be able to keep this value an Ipv4Addr

in order to handle this and wrap the logic in a reusable way clap provides an arg attribute `value_parser = parse_ip_address`

The fn signature would look like this
`parse_ip_address(input: &str) -> Result<Ipv4Addr>`

Reply:
I thought the `Ipv4Addr` type supported only the IP address itself and not the port. Wouldn't this end up requiring the port separately anyway?

Reply:
> Since it's just a processing function that converts A -> B,C, I'd just extract that logic into a private function on the command's file and write a unit test for that. You only have to ensure that the argument is parsed correctly.

I've gone ahead with this approach because of the situation I mentioned in the last comment.
I've also undrafted this PR since I've managed to test the parameter parsing with a local kafka installation and it seems to be working OK as of today. 

Let me know if you need any further touch-ups ✌️!

Reply:
@nidnogg I think @CAOakleyII was referring to `SocketAddr`, which represents an `Ipv4Addr` + a port. Using that you can avoid returning the ip/port split in two variables at `parse_bootstrap_server`.

Reply:
Thank you for that catch  @adrianbenavides  :)

I was indeed referring to `SocketAddr`

Reply:
Makes sense to me - I've modified it to use a `SocketAddr` as well as in the unit test, although the `start_kafka_service_impl` routine in `ockam_api/src/nodes/service/services.rs` [does use the port and IP in separate ways](https://github.com/build-trust/ockam/pull/4648/files#diff-9ec90b1c70033c4cc7a9648c1dfba47cdd60a45dce9b698e02de4c9ee1f32e3dR723-R724), so I kept it split there deeper in the flow. 

Let me know if you want to just use one parameter there as well!

Reply:
> Let me know if you want to just use one parameter there as well!

I think we can let that as it is 👍